internal/uvm/cpugroups.go

```diff
@@ -0,0 +1,121 @@
+package uvm
```

Comment:
I think we should move out the core CPU groups support into an `internal/cpugroup` package, and just call into that from `uvm.SetCPUGroup` and `uvm.UnsetCPUGroup`.

Reply:
There are some standalone pieces that I moved out, but the code that needs to be run in Set/Unset/Release require the uvm package, so I did not move those. Let me know if it looks okay.